phpunit/class-rest-blocks-controller-test.php

```diff
@@ -176,4 +176,35 @@ public function test_capabilities( $action, $role, $expected_status ) {
 			self::delete_user( $user_id );
 		}
 	}
+
+	/**
+	 * Check that the raw title and content of a block can be accessed by a
+	 * low-privileged role when no `context` param is provided.
+	 */
+	public function test_includes_raw_title_and_content() {
+		$user_id = $this->factory->user->create( array( 'role' => 'author' ) );
```

Comment:
User creation (and deletion at the end of this function) should be in the `::wpSetUpBeforeClass()` and `::wpTearDownAfterClass()` methods.

If the `::assertEquals()` call fails it will throw an exception, causing the test method to exit immediately, so the user wouldn't be properly deleted.

Reply:
I've moved user deletion to `::wpTearDownAfterClass`. The user we require is different for every test case so I've kept user creation as is.

Reply:
Oops, the fake user being different for every test case means we should be deleting it in `::tearDown()`, not `::wpTearDownAfterClass()`. Fixed in f8ebe5f72.

Reply:
It's fine to create a bunch of users in the setup and teardown methods, even when they're only used in one test. Kind of like what we do in [`WP_REST_Themes_Controller_Test`](https://github.com/WordPress/gutenberg/blob/master/phpunit/class-wp-rest-themes-controller-test.php).

It's easier to read if those methods are balanced, too: teardown just undoes whatever setup did.

Reply:
👍 changed in 4394eeb99.